test/datasets/test_multi30k.py

```diff
@@ -0,0 +1,82 @@
+import os
+import random
+import string
+import tarfile
+from collections import defaultdict
+from unittest.mock import patch
+
+from ..common.parameterized_utils import nested_params
+from torchtext.datasets import Multi30k
+
+from ..common.case_utils import TempDirMixin, zip_equal
+from ..common.torchtext_test_case import TorchtextTestCase
+
+
+def _get_mock_dataset(root_dir):
+    """
+    root_dir: directory to the mocked dataset
+    """
+    base_dir = os.path.join(root_dir, "Multi30k")
+    temp_dataset_dir = os.path.join(base_dir, "temp_dataset_dir")
+    os.makedirs(temp_dataset_dir, exist_ok=True)
+
+    seed = 1
+    mocked_data = defaultdict(list)
+    for file_name in ("train.de", "train.en", "val.de", "val.en", "test.de", "test.en"):
+        txt_file = os.path.join(temp_dataset_dir, file_name)
+        with open(txt_file, "w") as f:
+            for i in range(5):
+                rand_string = " ".join(
+                    random.choice(string.ascii_letters) for i in range(seed)
+                )
```

Comment:
One thought: since all of our datasets are utf-8 files, does it make sense to write unicode strings to make sure we don't have lingering bugs from default encodings when opening files? Maybe this is overkill, but it's been a big source of bugs when I did mostly windows development.

Reply:
Ya, agreed. I think it is a good suggestion!

Reply:
I will keep it as a follow-up item as generating random UTF-8 string is not trivial.